<div id="biosurfpipeline">

<table class="center" style="width:90%" >
<tr><td>
<div class=example"><p class=center><img src="images/biosurf-pipeline.png" width="690" height="796" border="1" usemap="#map" /></p></div>

<map name="map">
<!-- #$-:Image map file created by GIMP Image Map plug-in -->
<!-- #$-:GIMP Image Map plug-in by Maurits Rijk -->
<!-- #$-:Please do not edit lines starting with "#$" -->
<!-- #$VERSION:2.3 -->
<!-- #$AUTHOR:Jorge Oliveira -->
<area shape="rect" coords="254,16,427,71" alt="Go to user input" href="#input" />
<area shape="rect" coords="256,121,427,174" alt="Go to quality control" href="#qc" />
<area shape="rect" coords="471,129,525,145" alt="Go to FastQC Website" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" />
<area shape="rect" coords="471,146,552,163" alt="Go to Fastx Toolkit Website" href="http://hannonlab.cshl.edu/fastx_toolkit/" />
<area shape="rect" coords="63,246,203,313" alt="Go to Taxonomy and Function" href="#tax_func" />
<area shape="rect" coords="475,248,629,315" alt="Go to ORF Find" href="#gene" />
<area shape="rect" coords="51,376,218,424" alt="Go to Phylogenetic Analysis" href="#phylo" />
<area shape="rect" coords="74,432,204,482" href="http://huttenhower.sph.harvard.edu/metaphlan" />
<area shape="rect" coords="51,485,217,538" alt="Go to Alignment" href="#alignment" />
<area shape="rect" coords="111,551,154,566" alt="Go to BLAST website" href="http://blast.ncbi.nlm.nih.gov/Blast.cgi" />
<area shape="rect" coords="112,568,145,582" alt="Go to LAST website" href="http://last.cbrc.jp" />
<area shape="rect" coords="51,594,218,643" alt="Go to Database" href="#database" />
<area shape="rect" coords="113,657,162,677" alt="Go to RefSeq website" href="http://www.ncbi.nlm.nih.gov/refseq/" />
<area shape="rect" coords="112,680,186,713" alt="Go to BioSurfDB" href="#database" />
<area shape="rect" coords="467,375,640,424" alt="Go to Assembly" href="#assembly" />
<area shape="rect" coords="501,442,548,456" alt="Go to Velvet website" href="https://www.ebi.ac.uk/~zerbino/velvet/" />
<area shape="rect" coords="504,458,543,472" alt="Go to MIRA website" href="http://sourceforge.net/p/mira-assembler/wiki/Home/" />
<area shape="rect" coords="467,487,637,536" alt="Go to ORF Find" href="#orf" />
<area shape="rect" coords="511,553,604,569" alt="Go to MetaGeneMark" href="http://exon.gatech.edu/GeneMark/meta_gmhmmp.cgi" />
<area shape="rect" coords="512,570,603,584" alt="Go to FragGeneScan" href="http://omics.informatics.indiana.edu/FragGeneScan/" />
<area shape="rect" coords="257,643,425,693" alt="Go to Post-Processing" href="#post" />
<area shape="rect" coords="316,706,367,719" alt="Go to MEGAN5 website" href="http://ab.inf.uni-tuebingen.de/software/megan5/" />
<area shape="rect" coords="314,721,388,754" alt="Go to R project webpage" href="http://www.r-project.org" />
<area shape="rect" coords="316,755,362,770" alt="Go to STAMP website" href="http://kiwi.cs.dal.ca/Software/STAMP" />
</map>
<a id=input></a> 
<div class="titleLeft">User Input:</div>
<p>Here it is required a file in fastq format, which is similar to a FASTA file, but also contains sequence quality information. This format is required for the quality control programs to work. If no quality information is available, quality control must be bypassed.</p>
<p>&nbsp;</p>
<a id=qc></a>
<div class="titleLeft">Quality Control:</div>
<p> The first big problem to solve in metagenomics data analysis is the quality control. This step is crutial as low-quality reads or contaminations will largely influence the results.</p>
<p> Quality control is probably the step that requires the most interaction from the user. Based on the FASTQ quality report, the user must trim the contaminations, filter low-quality reads and watch for GC content problems.
<p>&nbsp;</p>
<a id=tax_func></a>
<div class="titleLeft">Taxonomy and Function:</div>
<p>In metagenomics, many of the projects aims to simply get a taxonomic analysis. In these cases, we have came to the conclusion that assembly should not be done, because <em>de novo</em> assemblers usually try to merge as many sequences as possible, resulting in a great loss of taxonomic diversity and abundance information.</p>
<p>&nbsp;</p>
<a id=phylo></a>
<div class="titleLeft">Phylogenetic Analysis:</div>
<p>For quick phylogenetic analysis, we recommend MetaPhlan, a software published in Nature, that uses a custom database to quickly and not demanding a lot of hardware computes phylogeny, creates heatmaps and hierarquical clusters and is thus a good starting hint on what is on your samples. </p>
<p>&nbsp;</p>
 <a id=alignment></a>
<div class="titleLeft">Alignment:</div>
<p>This is the step where basicaly we try to discover what are our sequencies. We try to discover if there is a similar sequence in a public or private, global or custom databse.</p>
<p>Alignment is often one of the headaches when in comes to process your metagenomes. Both the input data and the databases are growing steadly due to the advances in sequencing. It is crucial to choose the right software or you will be waiting weeks for your results.</p>
<p>We usually use BLAST when sensitivity is important and we are using a custom (smaller) database. When using a very large database (RefSeq, e.g.) we use LAST, a tool that deals more efficiently with very heterogenious inputs (metagenomes) generating much faster results.</p>
<p>&nbsp;</p>
<a id=database></a>
<div class="titleLeft">Database:</div>
<p>Another very important factor is which database to use. Once again if your goal is just to get a taxonomic profile, it might be apropriate (because it is much faster) to just download a 16S database from RefSeq. In fact, many of the sequencing kits used in metagenomics, are 16S targeted, therefore, it does not make sense to use other database in these cases.</p>
<p>Another option, if you are working with a focus on a specific domain (in biodegradation or surfactant production e.g.) it might be interresting to narrow the search to these sequences, using a custom database, like BioSurfDB, this is also much faster and allows the use of more strict and sensitive parameters in the alignment.</p>
<p>&nbsp;</p>
<a id=gene></a>
<div class="titleLeft">Gene Find and Caracterization:</div>
<p>The second possible objective we are covering is the gene find and caracterization. The main diference in this pipeline is the assembly step, that tries to find contiguous sequences and merge then (into Contigs). These contigs can have thousands of nucleotides and can represent a cds, gene, operon, chromossome, etc.</p>
<p>&nbsp;</p>
<a id=assembly></a>
<div class="titleLeft">Assembly:</div>
<p>After testing several <em>de novo</em> assemblers, we started using MIRA in a regular basis, mainly because it suports and works relatively well in all platforms. Unfortunaly it requires a lot of computational power (Up to 100GBs of RAM and rather 16 or more processing threads).</p>
<p>If you do not have this computational power, Velvet also works very well in most cases, despite being developed for 454 sequencing.</p>
<p>&nbsp;</p>
<a id=orf></a>
<div class="titleLeft">ORF Find:</div>
<p>Once again the main purpose is to discover what is in our sequences. But here the focus is not exactly measure the abundance, but identify the complete genomic or proteic sequence in order to characterize it. Here the softwares tipicaly try to find open reading frames and compare those frames with a database..</p>
<p>&nbsp;</p>
<a id=post></a>
<div class="titleLeft">Post-Processing and Visualization:</div>
<p>If one wishes to compare a great number of samples, or simply top perform a fast representation of the results, MEGAN is good option. It imports alignment results (in BLAST format), CSV tables, etc., and contains several graphics for taxonomic/functional assays (barplots, clouds, pie, cladograms, rarefaction curves, PCA, clustering), inclusively an option to map the aligned reads to a KEGG, COG or SEED map, allowing an interactive functional study.</p>
<p>Unfortunally, MEGAN is also very demanding computationally, specially for large datasets, making it dificult to use in a personal computer. The alternative is to use it in a server with display (real, virtual or with forwarding). MEGAN also has a "language" that allows some automation of processing, but might be dificult for begginers.</p>
<p>Statistics is mandatory for a serious (publishable) study. Therefore, statistical analysis, usually pairwise tests for significant differences are performed. For begginers STAMP might be perfect. It accepts TSV (tab separated values) tables, that can be created in MEGAN or many other softwares, and based on your data, STAMP recommends several statistical tests and creates barplots, error plots, PCA, etc.</p>
<p>For a larger project envolving many samples, it might be a good investment to learn some R. Its allows the creation of scripts to automate these tests.</p>
<p>&nbsp;</p>
</td></tr>
</table>
</div> <!-- end of biosurfpipine-->
